### PR TITLE
Change masking to direct labeling for TPU support.

### PR DESCRIPTION
**Pull request by srush**
Torch XLA conversion is very sensitive to certain core pytorch operations. These result in TPU being slower than CPU operation. The most obvious are binary masking and calls to item(). 

https://github.com/pytorch/xla/blob/master/TROUBLESHOOTING.md

This PR replaces some of these calls that are in the example/NER pathway.

### CODE REVIEW

File: src/transformers/modeling_bert.py [hunk 1]
```diff
@@ -1394,12 +1394,13 @@ def forward(
 
         outputs = (logits,) + outputs[2:]  # add hidden states and attention if they are here
         if labels is not None:
-            loss_fct = CrossEntropyLoss()
+            ignore_index = -100
+            loss_fct = CrossEntropyLoss(ignore_index=ignore_index)
```
Comment (LysandreJik) [inline src/transformers/modeling_bert.py]: Is there a reason for explicitly specifying the ignored index? Up to now, our philosophy was to let the default be PyTorch's default.
Explicitly specifying it means that if PyTorch ever changes the default, our losses will be incoherent with PyTorch's official losses.

Comment (srush) [reply src/transformers/modeling_bert.py]: Cool, I switched to that.

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @srush !
